lnwallet/channel.go

```diff
@@ -6355,6 +6355,53 @@ func (lc *LightningChannel) addHTLC(htlc *lnwire.UpdateAddHTLC,
 		return 0, err
 	}
 
+	// Let's check if this is a custom channel HTLC.
+	if lc.isCustomChannel() && htlc.CustomRecords != nil {
+
+		// If the htlc amount is below dust we will override it and set
+		// it to be the dust limit.
+		if btcutil.Amount(htlc.Amount/1000) <
+			lc.channelState.LocalChanCfg.DustLimit {
+
+			pd.Amount = lnwire.MilliSatoshi(
+				lc.channelState.LocalChanCfg.DustLimit * 1000,
+			)
+		}
+
+		// Construct the payment descriptor that will be used to append
+		// to the remote log.
+		remotePd := &PaymentDescriptor{
```

Comment:
not sure if this should also be inside the if body above?

I suppose the compensation only makes sense for dust, because if the sender defined an above-dust amount then maybe they actually want to forward that amount to the network.